vagrant/provisioning/roles/arkcase-app/tasks/main.yml

```diff
@@ -798,7 +798,31 @@
       application.rolesToGroups:
         {{ roles_to_groups_yaml | to_nice_yaml | indent(2) }}
   when: arkcase_version == "" or arkcase_version is version('3.3.3', '>=')
-    
+
+- name: Make sure AWS credentials are insterted in the arkcase-server.yaml file if they are defined
+  become: yes
+  become_user: arkcase
+  lineinfile:
+    path: /opt/arkcase/data/arkcase-home/.arkcase/acm/acm-config-server-repo/arkcase-server.yaml
+    state: present
+    backup: True
+    backrefs: True
+    regexp: '^awsAccessKeyId'
+    line: 'awsAccessKeyId: {{ awsAccessKeyId_fact }}'
+  when: awsAccessKeyId_fact is defined
+
+- name: Make sure AWS credentials are insterted in the arkcase-server.yaml file if they are defined
```

Comment:
Since these two lines are already present in the server.yml template, do you really need these two `lineinfile` tasks?  If so, do you need to make sure they appear immediately after `aws.credentials:`, and do you need to make sure `aws.credentials:` itself is present?